rclcpp/include/rclcpp/client.hpp

```diff
@@ -830,6 +919,11 @@ class Client : public ClientBase
       CallbackInfoVariant>>
   pending_requests_;
   std::mutex pending_requests_mutex_;
+
+private:
+  using ClientIntraProcessT = rclcpp::experimental::ClientIntraProcess<ServiceT>;
+  std::shared_ptr<ClientIntraProcessT> client_intra_process_;
```

Comment:
to match subscriptions implementation, we should have that `client_intra_process_` is a protected member in `client_base.hpp`.
Its type will be `ClientIntraProcessBase` and the `using` definition can be moved to the function that creates the object

Reply:
Done in [bec9017](https://github.com/ros2/rclcpp/pull/1847/commits/bec90173ddc2a273580308bda1fc27ac7c41e3c9)